pkg/ingester/ingester.go

```diff
@@ -577,6 +578,25 @@ func (i *Ingester) removeShutdownMarkerFile() {
 func (i *Ingester) loop() {
 	defer i.loopDone.Done()
 
+	// Delay the first flush operation by up to 0.8x the flush time period.
+	// This will ensure that multiple ingesters started at the same time do not
+	// flush at the same time. Flushing at the same time can cause concurrently
+	// writing the same chunk to object storage, which in AWS S3 leads to being
+	// rate limited.
+	jitter := time.Duration(rand.Int63n(int64(float64(i.cfg.FlushCheckPeriod.Nanoseconds()) * 0.8)))
+	initialDelay := time.NewTimer(jitter)
+	defer initialDelay.Stop()
```

Comment:
I think we need to do this slightly differently. I think the jitter needs to be on each evaluation such that the flush loop runs at an interval of say approx ever 30s +/- maybe 5s?

The reason being: the example I looked at the other day, the ingesters actually started maybe 10 minutes apart however their flush loops just unluckily ended up synchronized. So adding just an initial jitter would have the same probability of unluckiness which is apparently pretty high as we see this somewhat often.

Reply:
I think the jitter needs to be applied constantly, if it's only ever applied once I think we end up with the same odds of flush loops being synchronized and stuck that way indefinitely until an ingester restarts.

Reply:
Additionally to the initial delay, I replaced the Ticker with a ticker with jitter, see line 600ff